lib/capybara/cuprite/browser.rb

```diff
@@ -257,6 +268,22 @@ def attach_page(target_id = nil)
         target.page = Page.new(target.client, context_id: target.context_id, target_id: target.id)
         target.page
       end
+
+      def frame_obscured_at?(x, y)
+        frame = page.active_frame
+        return false if frame.main?
+
+        parent_frame = page.frame_by(id: frame.parent_id)
+        frame_node = parent_frame&.at_xpath("//iframe[@id=\"#{frame.name}\"]")
+        return false unless frame_node
```

Comment:
I'm not particularly happy with this code and would welcome suggestions. As mentioned in the PR, it would be great if this functionality (finding a `Ferrum::Node` for a given `Ferrum::Frame`) could be upstreamed into Ferrum.

Reply:
I've managed to replace this code with `frame.evaluate("window.frameElement")`, which is much nicer. There's now an upstream PR adding support for `Frame#node` here: https://github.com/rubycdp/ferrum/pull/524